azure-pipelines-yarn-template.yml

```diff
@@ -0,0 +1,26 @@
+jobs:
+- job: ${{ parameters.name }}
+  pool:
+    vmImage: ${{ parameters.vmImage }}
+  strategy:
+    matrix:
+      node_11_x:
+        node_version: 11.x
+      node_10_x:
+        node_version: 10.x
+      node_9_x:
+        node_version: 9.x
+      node_8_x:
+        node_version: 8.x
+      node_6_x:
+        node_version: 6.x
+    maxParallel: 5
+  steps:
+  - task: NodeTool@0
+    inputs:
+      versionSpec: $(node_version)
+    displayName: Install Node.js
+  - bash: yarn install --ignore-engines
```

Comment:
why this change? Is this causing some error?

Reply:
Because yarn strictly checks if the engines are compatible. Some packages weren't compatible. Try it out yourself.

Reply:
I’m not able to repro locally on node 8 or 10.

Reply:
Here from the build with Node 6

```
2018-11-18T09:40:14.5092272Z ##[section]Starting: Install dependencies
2018-11-18T09:40:14.5096081Z ==============================================================================
2018-11-18T09:40:14.5096251Z Task         : Bash
2018-11-18T09:40:14.5096355Z Description  : Run a Bash script on macOS, Linux, or Windows
2018-11-18T09:40:14.5096500Z Version      : 3.142.2
2018-11-18T09:40:14.5096632Z Author       : Microsoft Corporation
2018-11-18T09:40:14.5096743Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613738)
2018-11-18T09:40:14.5096937Z ==============================================================================
2018-11-18T09:40:14.6551118Z Generating script.
2018-11-18T09:40:14.6563997Z Script contents:
2018-11-18T09:40:14.6565158Z yarn install
2018-11-18T09:40:14.6606089Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/d6f53f8a-15c5-423f-b546-2228b0837968.sh
2018-11-18T09:40:15.3744371Z yarn install v1.10.1
2018-11-18T09:40:15.4807487Z info No lockfile found.
2018-11-18T09:40:15.4955047Z [1/5] Validating package.json...
2018-11-18T09:40:15.4979162Z [2/5] Resolving packages...
2018-11-18T09:40:18.1027191Z warning standard > eslint > file-entry-cache > flat-cache > circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
2018-11-18T09:40:18.5789270Z [3/5] Fetching packages...
2018-11-18T09:40:31.0580769Z error on-net-listen@1.1.2: The engine "node" is incompatible with this module. Expected version ">=9.4.0 || ^8.9.4". Got "6.14.4"
2018-11-18T09:40:31.0588843Z error Found incompatible module
2018-11-18T09:40:31.0592743Z info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
2018-11-18T09:40:31.1283049Z ##[error]Bash exited with code '1'.
2018-11-18T09:40:31.1307411Z ##[section]Finishing: Install dependencies
```
Hosted in `autocannon`
```
yarn why v1.12.1
[1/4] Why do we have the module "on-net-listen"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "on-net-listen@1.1.2"
info Reasons this module exists
   - "autocannon" depends on it
   - Hoisted from "autocannon#on-net-listen"
Done in 0.40s.
```